Modified folder structure and added unit test for issue#12

Shyam-Sundar-Raju opened the pull request.

Shyam-Sundar-Raju committed:
Modified folder structure, added unit test and CI

Shyam-Sundar-Raju committed:
Added interface instead of import

Ashrockzzz2003 committed:
Use table driven tests for configs/ and helpers/ and run them in parallel. Remove custom test runner.

go test ./... -v already does the job of testing. No need for a separate test runner. So removed it.
Modified 2 modules to use table driven tests for easier readability in the future.
Added t.Parallel() to make the tests run in parallel for faster execution. Referred https://pkg.go.dev/testing\#T.Run

Ashrockzzz2003 commented:
@Shyam-Sundar-Raju @Abhinav-ark 
I'm working on something on top of this PR. Please don't push updates to this!

Ashrockzzz2003 committed:
Refactor codebase to promote test driven development

Grouped menu components into menu. Writing tests for that is a huge task but is possible now as we can fake the request client method. Finding a way to fake the huh form is the next step.

Ashrockzzz2003 reviewed (approved):
LGTM

What's pending is adding test cases for huh forms.
We're now testing the parsing logic of amrita_pyq with sample test cases by interface substitution. 

@Abhinav-ark need your review on this one! @Shyam-Sundar-Raju do see if you can add the remaining test cases.

Ashrockzzz2003 commented:
/bounty 200 @Shyam-Sundar-Raju
